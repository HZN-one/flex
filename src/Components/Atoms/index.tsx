export * from "./FAButton";
export * from "./Icon";
export * from "../Molecules/FMFormTextField";
export * from "./FATextField";
export * from "../Molecules/FMSelect";
export * from "./FACheckbox";
export * from "./FASwitch";
export * from "./FARadio";
export * from "./FAPagination";
export * from "./FAChip";
export * from "./FATooltip";
export * from "./FAFormControlLabel";
export * from "./FAFormControl";
