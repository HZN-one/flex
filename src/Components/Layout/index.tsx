export * from "./Initiator";
export * from "./FLSettingsNavbar";
