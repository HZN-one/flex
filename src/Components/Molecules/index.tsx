export * from "./FMAccordion";
export * from "./FMSearch";
export * from "./FMCard";
export * from "./FMStepper";
export * from "./FMTableSkeleton";
export * from "./FMTimeline";
export * from "./FMLabelValue";
export * from "./FMTabsMenu";
export * from "./FMInputAvatar";
