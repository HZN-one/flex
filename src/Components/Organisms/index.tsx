export * from "./FOCardWithImage";
export * from "./FOCardWithStepper";
export * from "./FOCardWithTextField";
export * from "./FODialogWithAlert";
export * from "./FOHeader";
export * from "./FOPaperRow";
export * from "./FOSideBar";
