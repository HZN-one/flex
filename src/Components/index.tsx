export * from './Atoms'
export * from './Layout'
