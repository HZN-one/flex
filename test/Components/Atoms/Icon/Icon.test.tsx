/**
 * @jest-environment jsdom
 */

import * as React from "react";
import "@testing-library/jest-dom";

import { FAIcon } from "Components";

import { fireEvent, render } from "@testing-library/react";

describe("FAIcon", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <FAIcon testID="icon-example">add_circle</FAIcon>
    );
    expect(container).toMatchSnapshot();
  });

  it("should increment number on click", () => {
    let number = 1;
    const { getByTestId } = render(
      <FAIcon
        testID="icon-example"
        onClick={() => {
          number += 1;
        }}
      />
    );

    fireEvent.click(getByTestId("icon-example"), new MouseEvent("click"));

    expect(number).toBe(2);
  });

  it("should render android icon by iconName property", () => {
    const { getByTestId } = render(
      <FAIcon testID="icon-example">android</FAIcon>
    );

    expect(`${getByTestId("icon-example").innerHTML}`).toMatch("android");
  });
});
