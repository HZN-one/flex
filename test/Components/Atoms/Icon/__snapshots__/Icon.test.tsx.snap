// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAIcon should match snapshot 1`] = `
<div>
  <span
    aria-hidden="true"
    class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
    data-testid="icon-example"
  >
    add_circle
  </span>
</div>
`;
